src/sensors/sensor.cpp

```diff
@@ -30,33 +30,75 @@ SensorStatus Sensor::getSensorState() {
 }
 
 void Sensor::setAcceleration(Vector3 a) {
+#if ESP32 && SENSOR_THREADING
+    xSemaphoreTake(updateMutex, portMAX_DELAY);
+#endif
     acceleration = a;
     newAcceleration = true;
+#if ESP32 && SENSOR_THREADING
+    xSemaphoreGive(updateMutex);
+#endif
 }
 
 void Sensor::setFusedRotation(Quat r) {
+#if ESP32 && SENSOR_THREADING
+    xSemaphoreTake(updateMutex, portMAX_DELAY);
+#endif
     fusedRotation = r * sensorOffset;
     bool changed = OPTIMIZE_UPDATES ? !lastFusedRotationSent.equalsWithEpsilon(fusedRotation) : true;
     if (ENABLE_INSPECTION || changed) {
         newFusedRotation = true;
         lastFusedRotationSent = fusedRotation;
     }
+#if ESP32 && SENSOR_THREADING
+    xSemaphoreGive(updateMutex);
+#endif
 }
 
 void Sensor::sendData() {
+#if ESP32 && SENSOR_THREADING
+    Quat lquat;
+    xSemaphoreTake(updateMutex, portMAX_DELAY);
     if (newFusedRotation) {
+        newFusedRotation = false;
+        lquat = fusedRotation;
+        xSemaphoreGive(updateMutex);
+    } else {
+        xSemaphoreGive(updateMutex);
+        return;
+    }
+
+    {
+        networkConnection.sendRotationData(sensorId, &fusedRotation, DATA_TYPE_NORMAL, calibrationAccuracy);
```

Comment:
```suggestion
        networkConnection.sendRotationData(sensorId, &lquat, DATA_TYPE_NORMAL, calibrationAccuracy);
```

Shouldn't it be like that?

Also can be combined my taking lquat out of the define and having the same sendRotationData be used.